Signal/src/ViewControllers/GifPicker/GifPickerViewController.swift

```diff
@@ -31,16 +31,16 @@ class GifPickerViewController: OWSViewController, UISearchBarDelegate, UICollect
 
     public weak var delegate: GifPickerViewControllerDelegate?
 
-    var thread: TSThread?
-    var messageSender: MessageSender?
+    let thread: TSThread
+    let messageSender: MessageSender
 
     let searchBar: UISearchBar
     let layout: GifPickerLayout
     let collectionView: UICollectionView
     var noResultsView: UILabel?
     var searchErrorView: UILabel?
     var activityIndicator: UIActivityIndicatorView?
-
+    var selectedCell: UICollectionViewCell?
```

Comment:
I think this should be `selectedImageInfo` or `selectedSendingRendition`, not `selectedCell`.  `UICollectionView` can reload its contents at any time including during download of the "sending rendition".  The selected `GifPickerCell` might be reused, making the "retry download" logic below unsafe.

Reply:
Good point. Currently we were only using it's presence as a boolean anyway, so I just changed it to a boolean `hasSelectedCell`.